src/main.ts

```diff
@@ -77,3 +44,62 @@ export async function run(): Promise<void> {
     }
   }
 }
+
+type Inputs = {
+  accessToken: string;
+  secrets: string[];
+  identityUrl: string;
+  apiUrl: string;
+};
+
+function readInputs(): Inputs {
+  const accessToken: string = core.getInput("access_token", {
+    required: true,
+  });
+  const secrets: string[] = core.getMultilineInput("secrets", {
+    required: true,
+  });
+  const baseUrl: string = core.getInput("base_url");
+  let identityUrl: string = core.getInput("identity_url");
+  let apiUrl: string = core.getInput("api_url");
+
+  if (baseUrl) {
+    if (!isValidUrl(baseUrl)) {
+      throw TypeError("input provided for base_url not in expected format");
+    }
+    identityUrl = baseUrl + "/identity";
+    apiUrl = baseUrl + "/api";
+  }
+  if (!isValidUrl(identityUrl)) {
+    throw TypeError("input provided for identity_url not in expected format");
+  }
+  if (!isValidUrl(apiUrl)) {
+    throw TypeError("input provided for api_url not in expected format");
+  }
+
+  return {
+    accessToken,
+    secrets,
+    identityUrl,
+    apiUrl,
+  };
+}
+
+async function getBitwardenClient(inputs: Inputs): Promise<BitwardenClient> {
+  const settings: ClientSettings = {
+    apiUrl: inputs.apiUrl,
+    identityUrl: inputs.identityUrl,
+    userAgent: "bitwarden/sm-action",
+    deviceType: DeviceType.SDK,
+  };
+
+  const logLevel = core.isDebug() ? LogLevel.Debug : LogLevel.Info;
```

Comment:
I have some concerns around enabling the ability to set the Log Level to Debug.

Reply:
reverted to Info